runtime/src/test/java/io/novafoundation/nova/runtime/extrinsic/visitor/impl/Common.kt

```diff
@@ -0,0 +1,132 @@
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.Event
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.MetadataFunction
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.Module
+import org.junit.Assert
+import org.mockito.Mockito
+import java.math.BigInteger
+
+fun createTestModuleWithCall(
+    moduleName: String,
+    callName: String
+): Module {
+    return Module(
+        name = moduleName,
+        storage = null,
+        calls = mapOf(
+            callName to MetadataFunction(
+                name = callName,
+                arguments = emptyList(),
+                documentation = emptyList(),
+                index = 0 to 0
+            )
+        ),
+        events = emptyMap(),
+        constants = emptyMap(),
+        errors = emptyMap(),
+        index = BigInteger.ZERO
+    )
+}
+
+fun createExtrinsic(
+    signer: AccountId,
+    call: GenericCall.Instance,
+    events: List<GenericEvent.Instance>
+) = ExtrinsicWithEvents(
+    extrinsic = Extrinsic.DecodedInstance(
+        signature = Extrinsic.Signature(
+            accountIdentifier = bindMultiAddress(MultiAddress.Id(signer)),
+            signature = null,
+            signedExtras = emptyMap()
+        ),
+        call = call,
+    ),
+    extrinsicHash = "0x",
+    events = events
+)
+
+fun extrinsicSuccess(): GenericEvent.Instance {
+    return mockEvent("System", "ExtrinsicSuccess")
+}
+
+fun extrinsicFailed(): GenericEvent.Instance {
+    return mockEvent("System", "ExtrinsicFailed")
+}
+
+fun mockEvent(moduleName: String, eventName: String, arguments: List<Any?> = emptyList()): GenericEvent.Instance {
+    val module = Mockito.mock(Module::class.java)
+    whenever(module.name).thenReturn(moduleName)
+
+    val event = Mockito.mock(Event::class.java)
+    whenever(event.name).thenReturn(eventName)
+
+    return GenericEvent.Instance(
+        module = module,
+        event = event,
+        arguments = arguments
+    )
+}
+
+fun mockCall(moduleName: String, callName: String, arguments: Map<String, Any?> = emptyMap()): GenericCall.Instance {
+    val module = Mockito.mock(Module::class.java)
+    whenever(module.name).thenReturn(moduleName)
+
+    val function = Mockito.mock(MetadataFunction::class.java)
+    whenever(function.name).thenReturn(callName)
+
+    return GenericCall.Instance(
+        module = module,
+        function = function,
+        arguments = arguments
+    )
+}
+
+class TestModuleMocker {
+
+    val testModule = createTestModuleWithCall(moduleName = "Test", callName = "test")
+
+    val testInnerCall = GenericCall.Instance(
+        module = testModule,
+        function = testModule.call("test"),
+        arguments = emptyMap()
+    )
+
+    val testEvent = mockEvent(testModule.name, "test")
+
+    operator fun component1(): GenericCall.Instance {
+        return testInnerCall
+    }
+
+    operator fun component2():GenericEvent.Instance {
+        return testEvent
+    }
+}
```

Comment:
Why did we describe this class and `createTestModuleWithCall` here?
It seems better to move it to the test directory

Reply:
It is in the test directory already